src/envoy/http/authn/http_filter.cc

```diff
@@ -78,6 +83,53 @@ FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap&, bool) {
     }
   }
 
+  int origins_size = 0;
+  // In POC, only inspect the first credential_rule
+  if (config_.credential_rules_size() > 0 &&
+      config_.credential_rules()[0].origins_size() > 0) {
+    origins_size = config_.credential_rules()[0].origins_size();
+  }
+  ENVOY_LOG(debug, "AuthenticationFilter: {} config.origins_size()={}",
+            __func__, origins_size);
+
+  int num_config = 0;
+  if (jwt_store_.find(IstioAuthn::ORIGIN_STORE) == jwt_store_.end()) {
+    ENVOY_LOG(debug, "AuthenticationFilter: {} no ORIGIN_STORE", __func__);
+  } else if (jwt_store_[IstioAuthn::ORIGIN_STORE].size() == 0) {
+    ENVOY_LOG(debug, "AuthenticationFilter: {} ORIGIN_STORE has size 0",
+              __func__);
+  }
+
+  if (jwt_store_.find(IstioAuthn::ORIGIN_STORE) != jwt_store_.end() &&
+      jwt_store_[IstioAuthn::ORIGIN_STORE].size() > 0) {
+    num_config = jwt_store_[IstioAuthn::ORIGIN_STORE].size();
+  }
+  ENVOY_LOG(debug, "AuthenticationFilter: {} num_config is {}", __func__,
+            num_config);
+
+  if (num_config > 0) {
```

Comment:
The key part is how to find the correct config/store with each policy step.   Please rebase your code with Diem change and implement this in AuthenticatorBase class.   

It is too simple to only get the first JWT

Reply:
Stay tuned. Will update the code to iterate through the credentials and find correct JWT configs when rebasing the code.

Reply:
@qiwzhang, @diemtvu I have rebased the code and made the changes mentioned in your review comments: 
-- The JWT authentication is now implemented in AuthenticatorBase class. 
-- Iterate through all credential rules and methods, as well as peers' methods.
-- Find the correct config/store with each policy step.